lib/src/protocol/v2/message.cc

```diff
@@ -0,0 +1,76 @@
+#include <cpp-pcp-client/protocol/v2/message.hpp>
+#include <cpp-pcp-client/protocol/v2/schemas.hpp>
+
+#define LEATHERMAN_LOGGING_NAMESPACE CPP_PCP_CLIENT_LOGGING_PREFIX".message"
+
+#include <leatherman/logging/logging.hpp>
+
+// TODO(ale): disable assert() once we're confident with the code...
+// To disable assert()
+// #define NDEBUG
+#include <cassert>
+
+namespace PCPClient {
+namespace v2 {
+
+namespace lth_jc  = leatherman::json_container;
+
+//
+// Message
+//
+
+// Constructors
+
+Message::Message(lth_jc::JsonContainer envelope) : envelope_(envelope)
+{ }
+
+Message::Message(const std::string& transport_msg) :
+    Message(lth_jc::JsonContainer(transport_msg))
+{ }
+
+static bool validate_data(lth_jc::JsonContainer envelope, Validator const& validator)
```

Comment:
By just using `JsonContainer::get` perhaps you could use a const ref for `envelope`.